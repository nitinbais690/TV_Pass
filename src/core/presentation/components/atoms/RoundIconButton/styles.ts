import { scale } from 'qp-common-ui';
import { StyleSheet } from 'react-native';
import { RoundIconButtonProps } from '.';
import { appFonts } from 'core/styles/AppStyles';

export const styles = (props: RoundIconButtonProps, appColors: any) => {
    return StyleSheet.create({
        containerStyle: {
            alignItems: 'center',
        },
        circle: {
            flex: 1,
            width: props.width,
            height: props.height,
            borderRadius: props.width / 2,
            shadowColor: appColors.shadow,
            shadowOffset: { width: 0, height: 4 },
            shadowRadius: props.width / 2,
            elevation: 4,
            justifyContent: 'center',
            alignItems: 'center',
        },
        downloadProgress: {},
        actionTextStyle: {
            color: '#ECECEC',
            marginTop: scale(8),
            fontSize: appFonts.xxxs,
            fontFamily: appFonts.primary,
            fontWeight: '600',
        },
    });
};
