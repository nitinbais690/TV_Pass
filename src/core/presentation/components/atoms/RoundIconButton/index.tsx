import React from 'react';
import { AccessibilityProps, View, Text, TouchableOpacity } from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import useAppColors from 'core/presentation/hooks/use-app-colors';
import { styles } from './styles';

export default function RoundIconButton(props: RoundIconButtonProps) {
    const appColors = useAppColors();
    const roundIconStyles = styles(props, appColors);

    return (
        <TouchableOpacity style={[props.style]} onPress={props.onPress} testID={props.testID}>
            <View style={roundIconStyles.containerStyle}>
                <LinearGradient
                    style={roundIconStyles.circle}
                    colors={[appColors.primaryVariant3, appColors.primaryVariant1]}
                    useAngle={true}
                    angle={180}
                    locations={[0.094, 1]}>
                    {props.svgIcon}
                </LinearGradient>
                {props.title && <Text style={roundIconStyles.actionTextStyle}>{props.title}</Text>}
            </View>
        </TouchableOpacity>
    );
}

export interface RoundIconButtonProps extends AccessibilityProps {
    width: number;
    height: number;
    svgIcon: any;
    title?: string;
    style?: {};
    onPress?: () => void;
    testID?: string;
}
