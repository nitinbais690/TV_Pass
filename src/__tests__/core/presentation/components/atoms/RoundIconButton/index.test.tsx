import React from 'react';
import { render, toJSON } from '@testing-library/react-native';
import { StyleSheet } from 'react-native';
import RoundIconButton from 'core/presentation/components/atoms/RoundIconButton';
import DownloadIcon from 'assets/images/ic_download.svg';

jest.mock('core/presentation/hooks/use-app-colors');

describe('roundIconButton', () => {
    const buttonSize: number = 40;
    const iconSize: number = 24;

    test('renders correctly', () => {
        const { container } = render(
            <RoundIconButton
                width={buttonSize}
                height={buttonSize}
                svgIcon={<DownloadIcon width={iconSize} height={iconSize} />}
                title={'Download'}
            />,
        );
        expect(toJSON(container)).toMatchSnapshot();
    });

    test('render with style', () => {
        const styles = StyleSheet.create({
            container: {
                padding: 16,
            },
        });

        const { container } = render(
            <RoundIconButton
                width={buttonSize}
                height={buttonSize}
                svgIcon={<DownloadIcon width={iconSize} height={iconSize} />}
                title={'Download'}
                style={styles.container}
            />,
        );
        expect(toJSON(container)).toMatchSnapshot();
    });
});
