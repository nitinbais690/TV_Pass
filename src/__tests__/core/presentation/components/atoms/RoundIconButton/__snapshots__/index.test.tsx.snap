// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`roundIconButton render with style 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    hasTVPreferredFocus={false}
    isTVSelectable={true}
    style={
      Object {
        "opacity": 1,
        "padding": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <BVLinearGradient
        angle={180}
        colors={
          Array [
            4282073158,
            4281151543,
          ]
        }
        endPoint={
          Object {
            "x": 0.5,
            "y": 1,
          }
        }
        locations={
          Array [
            0.094,
            1,
          ]
        }
        startPoint={
          Object {
            "x": 0.5,
            "y": 0,
          }
        }
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 20,
            "elevation": 4,
            "flex": 1,
            "height": 40,
            "justifyContent": "center",
            "shadowColor": "#40000000",
            "shadowOffset": Object {
              "height": 4,
              "width": 0,
            },
            "shadowRadius": 20,
            "width": 40,
          }
        }
        useAngle={true}
      >
        <SvgMock
          height={24}
          width={24}
        />
      </BVLinearGradient>
      <Text
        style={
          Object {
            "color": "#ECECEC",
            "fontFamily": "ProximaNova-Regular",
            "fontSize": 12,
            "fontWeight": "600",
            "marginTop": 11.657142857142857,
          }
        }
      >
        Download
      </Text>
    </View>
  </View>
</View>
`;

exports[`roundIconButton renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    hasTVPreferredFocus={false}
    isTVSelectable={true}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <BVLinearGradient
        angle={180}
        colors={
          Array [
            4282073158,
            4281151543,
          ]
        }
        endPoint={
          Object {
            "x": 0.5,
            "y": 1,
          }
        }
        locations={
          Array [
            0.094,
            1,
          ]
        }
        startPoint={
          Object {
            "x": 0.5,
            "y": 0,
          }
        }
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 20,
            "elevation": 4,
            "flex": 1,
            "height": 40,
            "justifyContent": "center",
            "shadowColor": "#40000000",
            "shadowOffset": Object {
              "height": 4,
              "width": 0,
            },
            "shadowRadius": 20,
            "width": 40,
          }
        }
        useAngle={true}
      >
        <SvgMock
          height={24}
          width={24}
        />
      </BVLinearGradient>
      <Text
        style={
          Object {
            "color": "#ECECEC",
            "fontFamily": "ProximaNova-Regular",
            "fontSize": 12,
            "fontWeight": "600",
            "marginTop": 11.657142857142857,
          }
        }
      >
        Download
      </Text>
    </View>
  </View>
</View>
`;
